### PR TITLE
[WIP] Add support for running wasm-snip

### LINKED ISSUE #158: Add support for running wasm-snip
See also https://github.com/xtuc/rust-plugin/issues/1 and #153 

Basically there are four main options:

* Should we snip the `std::panicking` code? yes/no

* Should we snip the `std::fmt` code? yes/no

* Here are some specific functions to snip. (list of zero or more strings)

* Here are some regex patterns of functions to snip. (list of zero or more strings)

### PR DESCRIPTION
**Pull request by csmoe**
Closes #158

Make sure these boxes are checked! 📦✅

- [X] You have the latest version of `rustfmt` installed and have your 
      cloned directory set to nightly
```bash
$ rustup override set nightly
$ rustup component add rustfmt-preview --toolchain nightly
```
- [X] You ran `rustfmt` on the code base before submitting
- [X] You reference which issue is being closed in the PR text

✨✨ 😄 Thanks so much for contributing to wasm-pack! 😄 ✨✨

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
hey @csmoe ! thanks so much for this- it's looking really good. so we have a release coming up on wednesday and i think, unfortunately we're not gonna be able to sneak this in because there's some outstanding questions re #160 about configuration. i'm gonna make this as blocked for the moment, but the second after we get the most recent release out, i'll turn my attention back to this and see if we can't land it sooner than later and add the configuration option later so that we don't have to run you through too many rebases 😅 thanks again for submitting and sorry for having to ask you to wait, but hopefully it shouldn't be too long ✨ 💁

File: src/error.rs [hunk 1]
```diff
@@ -55,3 +57,21 @@ impl From<toml::de::Error> for Error {
         Error::SerdeToml(e)
     }
 }
+
+impl From<parity_wasm::elements::Error> for Error {
+    fn from(error: parity_wasm::elements::Error) -> Self {
+        Error::Cli {
+            message: format!("{}", "There was an output Error"),
```
Comment (mgattozzi) [inline src/error.rs]: You could just put "There was an output Error".to_string() instead here and below

File: src/error.rs [hunk 2]
```diff
@@ -55,3 +57,21 @@ impl From<toml::de::Error> for Error {
...
+            stderr: format!("{}", error),
+        }
+    }
+}
+
+impl From<failure::Error> for Error {
```
Comment (mgattozzi) [inline src/error.rs]: Does something return a `failure::Error` type that this is needed?

File: src/command/snip.rs [hunk 3]
```diff
@@ -0,0 +1,42 @@
+use error::Error;
+use parity_wasm::elements::{self, Serialize};
+use wasm_snip;
+
+#[derive(Clone, Debug, StructOpt)]
+pub struct SnipOpitons {
```
Comment (Hywan) [inline src/command/snip.rs]:
```diff
- SnipOpitons
+ SnipOptions
```

File: src/command/snip.rs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    functions: Vec<String>,
+    #[structopt(long = "pattern", short = "p")]
+    patterns: Vec<String>,
+    #[structopt(long = "snip_rust_fmt_code")]
+    snip_rust_fmt_code: bool,
+    #[structopt(long = "snip_rust_panicking_code")]
+    snip_rust_panicking_code: bool,
+}
+
+impl Into<wasm_snip::Options> for SnipOpitons {
```
Comment (Hywan) [inline src/command/snip.rs]:
```diff
- SnipOpitons
+ SnipOptions
```

**Comment by nicksrandall** [comment]
@csmoe are you still planning to work on this? If you've run out of bandwidth, I'd be happy to help. I'd like to get this and wasm-opt integration in.

**Comment by csmoe** [comment]
@nicksrandall really thanks. just do whatever you wanna.

**Comment by ashleygwilliams** [comment]
this PR is quite outdated- i'm sorry about this :( would you mind if we closed? we still want this feature but i imagine it will be easier to start a new PR to accomplish this. let me know what you think!

**Comment by csmoe** [comment]
yep It's better to restart

**Pull request closed without merging**